apps/evm/lib/evm/stack.ex

```diff
@@ -98,16 +98,18 @@ defmodule EVM.Stack do
         iex> EVM.Stack.pop_n([1, 2, 3], 4)
         {[1, 2, 3], []}
   """
-  @spec pop_n(t, integer()) :: {[EVM.val()], t}
-  def pop_n(stack, 0), do: {[], stack}
+  @spec pop_n(t, integer(), [integer()]) :: {[EVM.val()], t}
+  def pop_n(stack, n, acc \\ [])
 
-  def pop_n([h | t], n) do
-    {a, b} = pop_n(t, n - 1)
+  def pop_n(stack, 0, acc), do: {acc, stack}
 
-    {[h | a], b}
-  end
+  def pop_n([], _, acc), do: {acc, []}
+
+  def pop_n([head | tail], n, acc) do
+    new_acc = acc ++ [head]
```

Comment:
If you're interested in speed improvements, then I think it would be better to do `[head | acc]` and then do an `Enum.reverse` when we finish recursing, 

```elixir
def pop_n(stack, 0, acc), do {Enum.reverse(acc), stack}
```

`++` has to change the entire linked list to add an element to the end.

Reply:
yeah, was typing the same comment :D